robot/impl/local_robot_test.go

```diff
@@ -4239,3 +4239,104 @@ func TestMaintenanceConfig(t *testing.T) {
 		test.That(t, sensorUnBlocked, test.ShouldNotBeNil)
 	})
 }
+
+func TestRemovingOfflineRemotes(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	ctx := context.Background()
+	cfg := config.Config{
+		Remotes: []config.Remote{
+			{
+				Name:     "remoteOffline",
+				Insecure: true,
+				Address:  "123.123.123.123",
+			},
+		},
+	}
+	r := setupLocalRobot(t, ctx, &cfg, logger)
+	time.Sleep(2 * time.Second)
+	localRobot := r.(*localRobot)
+	remotes := localRobot.manager.resources.FindNodesByAPI(client.RemoteAPI)
+	test.That(t, remotes[0].Name, test.ShouldEqual, "remoteOffline")
+	r.Reconfigure(ctx, &config.Config{})
+	time.Sleep((2 * time.Second))
+	remotes = localRobot.manager.resources.FindNodesByAPI(client.RemoteAPI)
+	test.That(t, remotes, test.ShouldBeEmpty)
+}
+
+func TestRemovingOfflineRemote(t *testing.T) {
+	logger, _ := logging.NewObservedTestLogger(t)
+	ctx := context.Background()
+
+	motorName := "remoteMotorFoo"
+	motorResourceName := resource.NewName(motor.API, motorName)
+
+	remoteCfg := &config.Config{
+		Components: []resource.Config{
+			{
+				Name:                motorName,
+				Model:               resource.DefaultModelFamily.WithModel("fake"),
+				API:                 motor.API,
+				ConvertedAttributes: &fakemotor.Config{},
+			},
+		},
+	}
+
+	remoteRobot := setupLocalRobot(t, ctx, remoteCfg, logger.Sublogger("remote"))
+	remoteOptions, _, remoteAddr := robottestutils.CreateBaseOptionsAndListener(t)
+	err := remoteRobot.StartWeb(ctx, remoteOptions)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Set up a local main robot which is connected to the remote.
+	mainRobotCfg := &config.Config{
+		Remotes: []config.Remote{
+			{
+				Name:    "remote",
+				Address: remoteAddr,
+				// These values dictate how quickly we'll observe the remote going offline. And how
+				// quickly we'll observe it coming back online.
+				ConnectionCheckInterval: 10 * time.Millisecond,
+				ReconnectInterval:       10 * time.Millisecond,
+			},
+		},
+	}
+	mainRobotI := setupLocalRobot(t, ctx, mainRobotCfg, logger.Sublogger("main"))
+	// We'll manually access the resource manager to move the test forward.
+	mainRobot := mainRobotI.(*localRobot)
+	mainOptions, _, mainAddr := robottestutils.CreateBaseOptionsAndListener(t)
+	mainRobot.StartWeb(ctx, mainOptions)
+
+	// Create an "application" client to the robot.
+	mainClient, err := client.New(ctx, mainAddr, logger.Sublogger("client"))
```

Comment:
is the main client part of this test useful? seems like for the most part we're just interested in main -> remote not client -> main -> remote